src/backends/smtlib2.rs

```diff
@@ -130,6 +133,32 @@ impl<L: Logic> SMTLib2<L> {
     pub fn new_const<T: Into<L::Fns>>(&mut self, cval: T) -> NodeIndex {
         self.gr.add_node(cval.into())
     }
+
+    pub fn generate_asserts(&self) -> String {
+        // Write out all variable definitions.
+        let mut decls = Vec::new();
+        for (name, val) in &self.var_map {
+            let ni = &val.0;
+            let ty = &val.1;
+            if self.gr[*ni].is_var() {
+                decls.push(format!("(declare-fun {} () {})\n", name, ty));
+            }
+        }
+        // Identify root nodes and generate the assertion strings.
+        let mut assertions = Vec::new();
+        for idx in self.gr.node_indices() {
+            if self.gr.edges_directed(idx, EdgeDirection::Incoming).collect::<Vec<_>>().is_empty() {
+                if self.gr[idx].is_fn() && !self.gr[idx].is_bool() {
```

Comment:
@sushant94 the `self.gr[idx].is_bool()` check was added by you which made the tests to fail. The current change is not present in master. I have fixed it according to my understanding. Can you have a look? I could not seem to understand the need.